Mathlib/MeasureTheory/Measure/AddContent.lean

```diff
@@ -0,0 +1,184 @@
+/-
+Copyright (c) 2024 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Peter Pfaffelhuber
+-/
+import Mathlib.MeasureTheory.SetSemiring
+
+/-!
+# Additive Contents
+
+An additive content `m` on a set of sets `C` is a set function with value 0 at the empty set which
+is finitely additive on `C`. That means that for any finset `I` of pairwise disjoint sets in `C`
+such that `⋃₀ I ∈ C`, `m (⋃₀ I) = ∑ s ∈ I, m s`.
+
+Mathlib also has a definition of contents over compact sets: see `MeasureTheory.Content`.
+A `Content` is in particular an `AddContent` on the set of compact sets.
+
+TODO: refactor `Content` to use properties of `AddContent`.
+
+## Main definitions
+
+* `MeasureTheory.AddContent C`: additive contents over the set of sets `C`.
+
+## Main statements
+
+Let `m` be an `AddContent C`. If `C` is a set semi-ring (`IsSetSemiring C`) we have the properties
+
+* `MeasureTheory.sum_addContent_le_of_subset`: if `I` is a finset of pairwise disjoint sets in `C`
+  and `⋃₀ I ⊆ t` for `t ∈ C`, then `∑ s in I, m s ≤ m t`.
+* `MeasureTheory.addContent_mono`: if `s ⊆ t` for two sets in `C`, then `m s ≤ m t`.
+* `MeasureTheory.addContent_union'`: if `s, t ∈ C` are disjoint and `s ∪ t ∈ C`,
+  then `m (s ∪ t) = m s + m t`.
+  If `C` is a set ring (`IsSetRing`), then `addContent_union` gives the same conclusion without the
+  hypothesis `s ∪ t ∈ C` (since it is a consequence of `IsSetRing C`).
+
+If `C` is a set ring (`MeasureTheory.IsSetRing C`), we have, for `s, t ∈ C`,
+
+* `MeasureTheory.addContent_union_le`: `m (s ∪ t) ≤ m s + m t`
+* `MeasureTheory.addContent_le_diff`: `m s - m t ≤ m (s \ t)`
+
+-/
+
+open Set Finset
+
+open scoped ENNReal BigOperators
+
+namespace MeasureTheory
+
+variable {α : Type*} {C : Set (Set α)} {s t : Set α} {I : Finset (Set α)}
+
+/-- An additive content is a set function with value 0 at the empty set which is finitely additive
+on a given set of sets. -/
+structure AddContent (C : Set (Set α)) where
+  /-- The value of the content on a set. -/
+  toFun : Set α → ℝ≥0∞
+  empty' : toFun ∅ = 0
```

Comment:
can't this property be deduced from `sUnion'` (by taking the empty set)? If so, remove it from the definition, and prove it separately.

Reply:
That only works if the empty set is in C. I'd rather not have conditions on C in the definition of AddContent.